WeakAurasOptions/WeakAurasOptions.lua

```diff
@@ -3056,59 +3021,23 @@ function WeakAuras.ReloadTriggerOptions(data)
       WeakAuras.ReloadTriggerOptions(data);
     end
 
-    removeFuncs(displayOptions[id]);
-
+    trigger_options = union(trigger_options, WeakAuras.GetGenericTriggerOptions(data));
     if(optionTriggerChoices[id] >= 1 and getAll(data, {"trigger", "type"}) == "aura") then
-      local aura_options = WeakAuras.GetBuffTriggerOptions(data, trigger);
-      displayOptions[id].args.trigger.args = union(trigger_options, aura_options);
```

Comment:
The overall code got simpler by assinging to trigger_options here and doing the assignment for displayOptions[id].args.trigger.args below.